src/GraphQL/Types/Composite/AutoRegisteringObjectGraphType.cs

```diff
@@ -51,54 +52,74 @@ public AutoRegisteringObjectGraphType(params Expression<Func<TSourceType, object
+                    include = attr.ShouldInclude(memberInfo, false);
                     if (!include)
                         break;
                 }
                 if (!include)
                     continue;
-                var fieldType = CreateField(propertyInfo);
+                var fieldType = CreateField(memberInfo);
                 if (fieldType != null)
                     yield return fieldType;
             }
         }
 
         /// <summary>
-        /// Processes the specified property and returns a <see cref="FieldType"/>.
-        /// May return <see langword="null"/> to skip a property.
+        /// Processes the specified member and returns a <see cref="FieldType"/>.
+        /// May return <see langword="null"/> to skip a member.
         /// </summary>
-        protected virtual FieldType? CreateField(PropertyInfo propertyInfo)
+        protected virtual FieldType? CreateField(MemberInfo memberInfo)
         {
-            var typeInformation = GetTypeInformation(propertyInfo);
+            var typeInformation = GetTypeInformation(memberInfo);
             var graphType = typeInformation.ConstructGraphType();
-            return AutoRegisteringHelper.CreateField(propertyInfo, graphType, false);
+            return AutoRegisteringHelper.CreateField(memberInfo, graphType, false);
         }
 
         /// <summary>
-        /// Returns a list of properties that should have fields created for them.
+        /// Returns a list of properties, methods or fields that should have fields created for them.
+        /// <br/><br/>
+        /// Unless overridden, returns a list of public instance readable properties (including properties declared on
+        /// inherited classes) and public instance methods (excluding methods declared on inherited classes) that
+        /// do not return <see langword="void"/> or <see cref="Task"/>.
         /// </summary>
-        protected virtual IEnumerable<PropertyInfo> GetRegisteredProperties()
-            => AutoRegisteringHelper.ExcludeProperties(
+        protected virtual IEnumerable<MemberInfo> GetRegisteredMembers()
+        {
+            var props = AutoRegisteringHelper.ExcludeProperties(
                 typeof(TSourceType).GetProperties(BindingFlags.Public | BindingFlags.Instance).Where(x => x.CanRead),
                 _excludedProperties);
+            var methods = typeof(TSourceType).GetMethods(BindingFlags.Public | BindingFlags.Instance | BindingFlags.DeclaredOnly)
+                .Where(x =>
+                    !x.ContainsGenericParameters && //exclude methods with open generics
+                    !x.IsSpecialName &&             //exclude methods generated for properties
+                    x.ReturnType != typeof(void) && //exclude methods which do not return a value
+                    x.ReturnType != typeof(Task) && //exclude methods which do not return a value
+                    x.GetParameters().Length == 0); //exclude methods which contain arguments
```

Comment:
I guess it will be improved in the next PR?

Reply:
Yes